src/components/style.components.ts

```diff
@@ -52,12 +73,28 @@ export class StyleComponent implements OnDestroy, AfterContentInit {
 
     ngAfterContentInit() {
         this.update();
-        this.childSubscription$ = this.iconStyleDirective.onChanged.subscribe((): void => {
-            this.update();
-        });
+        if (this.iconStyleDirective) {
+            this.childSubscription$ = this.iconStyleDirective.onChanged.subscribe((): void => this.update());
+        }
+        else if (this.strokeStyleDirective) {
```

Comment:
Can stroke and icon not be used on the same style element?
If they can, this shouldn't be ```else if``` but only ```if```
same for the update function